### PR TITLE
Refactor `metrics.dependence` module

### PR DESCRIPTION
**Pull request by tsalo**
Closes none.

Changes proposed in this pull request:

- Add type hints to dependence metric functions.
- Require function calls to use keyword arguments.

### CODE REVIEW

**Review by eurunuela — APPROVED** [review]
LGTM!

File: tedana/metrics/collect.py [hunk 1]
```diff
@@ -119,7 +119,10 @@ def generate_metrics(
     metric_maps = {}
     if "map weight" in required_metrics:
         LGR.info("Calculating weight maps")
-        metric_maps["map weight"] = dependence.calculate_weights(data_optcom, mixing)
+        metric_maps["map weight"] = dependence.calculate_weights(
+            data_optcom=data_optcom,
```
Comment (tsalo) [inline tedana/metrics/collect.py:123]: Related to https://github.com/ME-ICA/tedana/pull/1086#discussion_r1574904654- we use `data_oc` in other places. I'd like to commit to one or the other and make the variable names consistent between this PR and #1086.

Comment (eurunuela) [reply tedana/metrics/collect.py:123]: I personally like `data_optcom` better.

Comment (tsalo) [reply tedana/metrics/collect.py:123]: That works for me. @handwerkerd do you have an opinion on this?

Comment (handwerkerd) [reply tedana/metrics/collect.py:152]: No opinion.

**Review by handwerkerd — APPROVED** [review]

**Pull request merged**